smart-contracts/contracts/UnlockUtils.sol

```diff
@@ -0,0 +1,63 @@
+pragma solidity 0.5.7;
+
+// This contract provides some utility methods for use with the unlock protocol smart contracts.
+// Borrowed from:
+// https://github.com/oraclize/ethereum-api/blob/master/oraclizeAPI_0.5.sol#L943
+
+contract UnlockUtils {
+
+  function strConcat(
+    string memory _a,
+    string memory _b
+  ) internal
+    pure
+    returns(string memory)
+ {
+    bytes memory _ba = bytes(_a);
+    bytes memory _bb = bytes(_b);
+    string memory ab = new string(_ba.length + _bb.length);
+    bytes memory bab = bytes(ab);
+    uint k = 0;
+    for(uint i = 0; i < _ba.length; i++) {
+      bab[k++] = _ba[i];
+    }
+    for(uint i = 0; i < _bb.length; i++) {
+      bab[k++] = _bb[i];
+    }
+    return string(bab);
+  }
+
+  function strConcat(
```

Comment:
These helpers are optional - but in the end I think we want 5 variables, so it would be:
tokenUri == baseUri + lockAddress + somethingInBetween + tokenId + extension
e.g. `https://unlock-protocol.com/metadata/0x1288...8181/101.json`

which we could do with `strConcat(strConcat(strConcat(strConcat(baseUri, lockAddress), somethingInBetween), tokenId), extension)`

Reply:
at the moment this is breaking CI

Reply:
Yeah, i see that. will fix it up.